shaders/galaxy_vert.glsl

```diff
@@ -1,18 +1,29 @@
 attribute vec4 in_Position;
-attribute vec4 in_TexCoord0;
+attribute vec2 in_TexCoord0;
+attribute float in_Size;
+attribute float in_ColorIndex;
+attribute float in_Brightness;
 
 uniform sampler2D colorTex;
 
+uniform mat4 m;
+uniform mat3 viewMat;
+
+uniform float size;
+uniform float brightness;
+
 varying vec4 color;
 varying vec2 texCoord;
 
 void main(void)
 {
-    // we pass color index as a normalized byte reusing in_TexCoord0.z
-    float t = in_TexCoord0.z;
-    // we pass alpha values as a normalized byte reusing in_TexCoord0.w
-    float a = in_TexCoord0.w;
-    color = vec4(texture2D(colorTex, vec2(t, 0.0)).rgb, a);
-    texCoord = in_TexCoord0.st;
-    set_vp(in_Position);
+    float s = size * in_Size;
+    vec4 p = m * in_Position;
+    float screenFrac = s / length(p);
```

Comment:
Original code uses check
```
if (screenFrac < 0.1f)
```
to add a new billboard.

Reply:
should we pass a value to fragment shader to let it discard? 

```
varying float valid;

if (screenFrac < 0.1f)
  valid = 1.0
else
  valid = -1.0
```

```
varying float valid;

if (valid < 0.0)
    discard;
```

Reply:
`discard` is supported with desktop GL since version 3, if i remember correctly. what you can do is to leave position as is, in that case you'll have _degenerated_ triangle (triangle with zero area) and hardware won't draw it.

Reply:
set `gl_Position = vec4(0.0, 0.0, 0.0, 1.0);` when `screenFrac >= 0.1`